### PR TITLE
Add `hatchling` as a build dependency

### PR DESCRIPTION
**Pull request by ofek**
### Motivation

Required for these https://github.com/DataDog/integrations-core/pulls?q=is%3Apr+is%3Aopen+author%3Aofek+Add+pyproject.toml+file

### CODE REVIEW

File: omnibus/config/software/datadog-agent-integrations-py3.rb [hunk 1]
```diff
@@ -128,9 +128,10 @@
     # Prepare the build env, these dependencies are only needed to build and
     # install the core integrations.
     #
+    command "#{pip} install hatchling==0.11.2"
     command "#{pip} install wheel==0.34.1"
     command "#{pip} install pip-tools==6.4.0"
-    uninstall_buildtime_deps = ['rtloader', 'click', 'first', 'pip-tools']
+    uninstall_buildtime_deps = ['rtloader', 'click', 'first', 'pip-tools', 'hatchling']
```
Comment (KSerrania) [inline omnibus/config/software/datadog-agent-integrations-py3.rb:134]: If I read the code correctly, we currently uninstall the `uninstall_buildtime_deps` after installing `datadog_checks_base` and `datadog_checks_downloader`, but before we install the integration wheels. If we need `hatchling` for all integration wheels, should we move [that uninstall section](https://github.com/DataDog/datadog-agent/blob/ee69aa446143aa8e9440465dde5a31afca950fe7/omnibus/config/software/datadog-agent-integrations-py3.rb#L214-L221) to after the build of integrations? In that case I think it should be the last thing we run in [this block](https://github.com/DataDog/datadog-agent/blob/ee69aa446143aa8e9440465dde5a31afca950fe7/omnibus/config/software/datadog-agent-integrations-py3.rb#L302-L393).

**Review by trishankatdatadog — APPROVED** [review]

**Review by KSerrania — APPROVED** [review]

File: omnibus/config/software/datadog-agent-integrations-py3.rb (hunk 1 above)
Comment (sarah-witt) [reply omnibus/config/software/datadog-agent-integrations-py3.rb:134]: Will we need the `uninstall_buildtime_deps` that aren't `hatchling` for installing each integration?

Comment (ofek) [reply omnibus/config/software/datadog-agent-integrations-py3.rb:134]: No

Comment (sarah-witt) [reply omnibus/config/software/datadog-agent-integrations-py3.rb:134]: Sure, makes sense

**Review by sarah-witt — APPROVED** [review]

**Pull request merged**